packages/back-end/src/app/controllers/easy-genomics/laboratory/update-laboratory.lambda.ts

```diff
@@ -37,14 +39,23 @@ export const handler: Handler = async (
     // Lookup by LaboratoryId to confirm existence before updating
     const existing: Laboratory = await laboratoryService.queryByLaboratoryId(id);
 
+    // TODO: check if lab not found
+
+    // Only Organisation Admins and Laboratory Managers are allowed to edit laboratories
+    if (
+      !validateOrganizationAdminAccess(event, existing.OrganizationId) ||
+      !validateLaboratoryManagerAccess(event, existing.OrganizationId, existing.LaboratoryId)
+    ) {
+      throw new UnauthorizedAccessError();
+    }
+
     const response: Laboratory = await laboratoryService
       .update(
         {
           ...existing,
           Name: request.Name,
           Description: request.Description,
           Status: 'Active',
-          S3Bucket: request.S3Bucket, // S3 Bucket Full Name
```

Comment:
I think you need to rebase your branch as this would removal would lead to a regression.